### PR TITLE
feat: Add support to clone an existing model on Butler

### PR DESCRIPTION
**Pull request by jayanthm**
### Related Issues
- fixes #issue-number

### Summary of Changes
Adds support to clone an existing pre-trained or custom model on Butler as a new custom model.
Also Adds the ability to debug pytest from VSCode

### Test Plan
pytest -v -m unit_tests

### Checklist
- [ ] PR title uses [conventional commit convention](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] Changes are properly tested
- [ ] Changes are properly documented

### CODE REVIEW

File: docai/models/model_utils.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from docai.generated.models.column_dto import ColumnDto
+from docai.generated.models.document_type_field_dto import DocumentTypeFieldDto
+from docai.generated.models.document_type_table_dto import DocumentTypeTableDto
+from docai.generated.models.field_dto import FieldDto
+from docai.generated.models.model_field_type import ModelFieldType
+from docai.generated.models.table_dto import TableDto
+
+
+def modelFieldToFieldDto(modelFieldDto: DocumentTypeFieldDto) -> FieldDto:
+    # DocumentTypeFieldDto doesn't support field types, assume everything is a text for now
```
Comment (butler-matt) [inline docai/models/model_utils.py:10]: Not sure I follow why DocumentTypeFieldDto doesn't include signatures/checkboxes etc. Where does that type come from?

File: docai/test/test_load_annotations.py [hunk 2]
```diff
@@ -1,28 +1,33 @@
 import logging
 import os
 
+import pytest
+
 from docai import AnnotationClient
 from docai.annotations.layoutlm_utils import normalize_ner_annotation_for_layoutlm
 
 # Run using 'python -m docai.test.test_load_annotations'
```
Comment (butler-matt) [inline docai/test/test_load_annotations.py:9]: Whit the changes we have made, is this still going to be the correct way to run this script? If not, can we update accordingly?

File: docai/models/model_client.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+from docai.common.http_utils import verify_response_or_raise
+from docai.generated.api.document_types import get_document_type_schema
+from docai.generated.api.models import create_custom_model, get_model, get_model_details
+from docai.generated.models import CreateModelDto, ModelInfoDto
+from docai.generated.models.document_type_schema_dto import DocumentTypeSchemaDto
+from docai.models.model_utils import modelFieldToFieldDto, modelTableToTableDto
+
+
+class ModelClient(BaseClient):
+    def clone_model(self, model_id: str, model_name: str) -> ModelInfoDto:
```
[the remark this answers is not in the document]
Comment (davidlizeng) [reply docai/models/model_client.py:14]: maybe clone_model_config or clone_model_schema?

File: docai/models/model_utils.py (hunk 1 above)
Comment (davidlizeng) [reply docai/models/model_utils.py:10]: The get_document_type_schema endpoint that Jay is using here is actually a legacy endpoint that was only used on the legacy training UI.

Longer term, we'd probably want to delete this legacy endpoint and adapt a newer endpoint that works across all doc extraction models.

File: docai/models/tests/test_model_client.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+                {"id": "b2adff10-a67d-4cb4-b8b7-ae2a7a91e86d", "name": "Deposit Amount"},
+                {"id": "914b0414-a3a9-48c6-80e4-fad5bf4e1397", "name": "Deposit Date"},
+                {"id": "9b6c2f94-bfbd-494a-b899-850216231f46", "name": "Deposit Description"},
+                {"id": "289a56c1-c828-475e-bca4-c046b8c61afe", "name": "Withdrawal Amount"},
+                {"id": "8fa40caf-17d9-4490-8b06-ea32e1a3d3fd", "name": "Withdrawal Date"},
+                {"id": "2b16a52d-61fd-418d-9a5d-7a91b05c1bb4", "name": "Withdrawal Description"},
+            ],
+        }
+    ],
+}
```
Comment (davidlizeng) [inline docai/models/tests/test_model_client.py:86]: What was the reason for the mock input being declared as a dictionary as opposed to creating a XYZDto object directly?

File: pyproject.toml [hunk 5]
```diff
@@ -61,8 +61,11 @@ line_length = 120
 include_trailing_comma = true
 
 [tool.pytest.ini_options]
+markers = [
+    "unit_tests: mark a test as one of the normal unit tests.",
+    "e2e_tests: mark a test as one of e2e tests.",
```
Comment (davidlizeng) [inline pyproject.toml:66]: very nice :)

File: docai/test/test_load_annotations.py (hunk 2 above)
Comment (davidlizeng) [reply docai/test/test_load_annotations.py:9]: The README.md has a section on testing, would be good to update that with instructions on how to run UTs and e2e tests.

File: docai/models/model_utils.py (hunk 1 above)
Comment (butler-matt) [reply docai/models/model_utils.py:10]: Why are we using that endpoint vs a new one?

File: docai/models/tests/test_model_client.py (hunk 4 above)
Comment (jayanthm) [reply docai/models/tests/test_model_client.py:86]: It's just a lot easier for the test writer to work with imo than instantiating an object. The flow I want to setup is they get an API response from curl/debugger/swagger ui. And they remove sensitive bits, and then use it elsewhere.

File: docai/test/test_load_annotations.py (hunk 2 above)
Comment (jayanthm) [reply docai/test/test_load_annotations.py:9]: ack, good idea. Will update

File: docai/models/model_client.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jayanthm) [reply docai/models/model_client.py:14]: going to go with `clone_model_schema`

File: docai/models/model_utils.py (hunk 1 above)
Comment (jayanthm) [reply docai/models/model_utils.py:10]: @butler-matt the old endpoint is the one that works for both PDTs and Custom Models. We can update this code once the new endpoint can also handle both

Comment (butler-matt) [reply docai/models/model_utils.py:10]: yuck :(

Comment (davidlizeng) [reply docai/models/model_utils.py:10]: Correct me if I'm wrong, but I think @jayanthm also has no intention of publishing this SDK yet, he's just using this codebase locally to push support scripts ahead a bit.

Overall seems low risk until we get a real endpoint that can support this.

**Review by davidlizeng — APPROVED** [review]

**Review by butler-matt — APPROVED** [review]

**Pull request merged**